### ISSUE TITLE
target-size doesn't respect inline exception for display: inline-block elements

### ISSUE THREAD

**Issue by dbjorge** [dequelabs/axe-core#4392]
### Product

axe-core

### Product Version

4.9.0

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation

The tooltip-button with id `repro` in the repro code below meets SC 2.5.8's Inline exception, so it shouldn't fail the `target-size` rule.

#### Actual
False positive (it does fail the rule)

#### How to Reproduce
Run the `target-size` rule against the following repro:

```html
<style>
  .no-spacing * {
    margin: 0;
    padding: 0;
  }
  .tooltip {
    border: 0;
    background: none;
    font-family: inherit;
    font-size: inherit;
    cursor: help;
    text-decoration: dotted underline;
  }
</style>
<div class="no-spacing">
  <p>This is a <button id="repro" class="tooltip">term</button> in a sentence.</p>
  <button style="width: 500px; height: 28px">this is a button that's within 24px of the term</button>
</div>
```

#### Additional context

This was reported as impacting a customer website by one of our internal auditors.

This is happening because the button in question is `display: inline-block` (the default for buttons), which the `widget-not-inline-matches` implementation used as `target-size`'s matcher treats as being "block-like" for the purposes of its `isInTextBlock` implementation (its code equivalent of the inline exception).

This is related to #3841, but in the other direction; in both cases, the `isInTextBlock` behavior that treats inline-blocks as not being part of surrounding text blocks is the root cause of the issues, but in #3841 that creates false negatives in some circumstances, and here it's instead creating false positives in some circumstances.

Like @WilcoFiers mentioned in https://github.com/dequelabs/axe-core/issues/3841#issuecomment-1373729367, because `inline-block` is the default display for buttons, it's difficult in practice for us to tell whether an `inline-block` should be treated as part of surrounding text or not (in the repro example above, both buttons are inline-block, but we'd ideally come up with different answers). I definitely don't think we should blanket-incomplete all inline-block things (that would make `target-size` treat most buttons as inapplicable), but maybe incompleting `inline-block` things that meet the rest of the `parentText`/`widgetText` comparison in `isInTextBlock` could make sense.

**Comment by MelSumner** [comment]
Came looking for this; per [2.5.8 Target Minimum](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html), targets in a sentence are acceptable exceptions.  


> Inline: The target is in a sentence or its size is otherwise constrained by the line-height of non-target text;

**Comment by straker** [comment]
To solve this we're thinking we'll look at the siblings of the `inline-block` node and if at least one sibling is inline or text then the inline-block node is potentially inline (this also applies to other inline block-like displays such as `inline-flex` and `inline-grid`). This means that `isInTextBlock` needs to be updated to return a 3rd indeterminate state (e.g. `undefined`) that signifies the potential to be inline. 

The caller of `isInTextBlock` then determines what to do with the indeterminate result. For `link-in-text-block` that means the matcher we will treat it as a block element and return false in the matcher to not return potential false positives. For `target-offset` we'll treat it as an inline element and ignore it as a potential neighbor for nearest neighbor calculations.

**Comment by padmavemulapati** [comment]
Verified with the latest dev code - now with the repro code snippet - 

```
<style>
  .no-spacing * {
    margin: 0;
    padding: 0;
  }
  .tooltip {
    border: 0;
    background: none;
    font-family: inherit;
    font-size: inherit;
    cursor: help;
    text-decoration: dotted underline;
  }
</style>
<div class="no-spacing">
  <p>This is a <button id="repro" class="tooltip">term</button> in a sentence.</p>
  <button style="width: 500px; height: 28px">this is a button that's within 24px of the term</button>
</div>
```
this is not failing `target-size` rule, reproduced that with the earlier version, `target-size` is failing.

https://github.com/user-attachments/assets/f13e19dd-7eab-4165-8021-bd498d6c76d0

## Environment:
| **Label**                         | **Value** |
| -------------------------- | ------ |
| **Product**                | _Axe-core_ |
| ** version**           | _4.11.2_ |
| **OS**                              | _Windows 11 & MAC-26.3_ |
| **Tested Browsers**       | _Chrome Version 147.0.7_ |
